sqlmesh/core/engine_adapter/base_postgres.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+import typing as t
+
+from sqlglot import exp
+
+from sqlmesh.core.engine_adapter.base import EngineAdapter
+from sqlmesh.core.engine_adapter.shared import DataObject, DataObjectType
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core._typing import TableName
+
+
+class BasePostgresEngineAdapter(EngineAdapter):
+    def columns(self, table_name: TableName) -> t.Dict[str, str]:
+        """Fetches column names and types for the target table."""
+        table = exp.to_table(table_name)
+        sql = (
+            exp.select("column_name", "data_type")
+            .from_("information_schema.columns")
+            .where(f"table_name = '{table.alias_or_name}' AND table_schema = '{table.args['db']}'")
+        )
+        self.execute(sql)
+        resp = self.cursor.fetchall()
+        return {column_name: data_type for column_name, data_type in resp}
+
+    def table_exists(self, table_name: TableName) -> bool:
+        """
+        Postgres doesn't support describe so I'm using what the redshift cursor does to check if a table
+        exists. We don't use this directly in order for this to work as a base class for other postgres
+
+        Reference: https://github.com/aws/amazon-redshift-python-driver/blob/master/redshift_connector/cursor.py#L528-L553
+        """
+        table = exp.to_table(table_name)
+
+        # Postgres doesn't support catalog
+        if table.args.get("catalog"):
+            return False
+
+        sql = (
```

Comment:
i see you copied me so that's good

Reply:
What PR/branch are you referring to where you made these changes/I copied?